src/android/usb_host/android_uvc.cpp

```diff
@@ -0,0 +1,1804 @@
+            ctrl->bmLayoutPerStream = QW_TO_QUAD(buf + 40);
+        }
+        //         else
+        //             ctrl->dwClockFrequency = devh->info->ctrl_if.dwClockFrequency;
+
+        /* fix up block for cameras that fail to set dwMax* */
+        if (ctrl->dwMaxVideoFrameSize == 0) {
+            uvc_frame_desc_t *frame = usbhost_uvc_find_frame_desc(devh, ctrl->bFormatIndex,
+                                                                  ctrl->bFrameIndex);
+
+            if (frame) {
+                ctrl->dwMaxVideoFrameSize = frame->dwMaxVideoFrameBufferSize;
+            }
+        }
+    }
+    return UVC_SUCCESS;
+}
+
+/** @brief Reconfigure stream with a new stream format.
+* @ingroup streaming
+*
+* This may be executed whether or not the stream is running.
+*
+* @param[in] strmh Stream handle
+* @param[in] ctrl Control block, processed using {usbhost_uvc_probe_stream_ctrl} or
+*             {uvc_get_stream_ctrl_format_size}
+*/
+uvc_error_t usbhost_uvc_stream_ctrl(usbhost_uvc_stream_handle_t *strmh, uvc_stream_ctrl_t *ctrl) {
+    uvc_error_t ret = UVC_SUCCESS;
+
+    if (strmh->stream_if->bInterfaceNumber != ctrl->bInterfaceNumber) {
+        return UVC_ERROR_INVALID_PARAM;
+    }
+
+    /* @todo Allow the stream to be modified without restarting the stream */
+    if (strmh->running) {
+        return UVC_ERROR_BUSY;
+    }
+
+    // Set streaming control block
+    ret = usbhost_uvc_query_stream_ctrl(strmh->devh, ctrl, 0, UVC_SET_CUR);
+    if (ret != UVC_SUCCESS) {
+        return ret;
+    }
+
+    strmh->cur_ctrl = *ctrl;
+    return UVC_SUCCESS;
+}
+
+static usbhost_uvc_stream_handle_t *
+usbhost_uvc_get_stream_by_interface(usbhost_uvc_device *devh, int interface_idx) {
+    usbhost_uvc_stream_handle_t *strmh;
+
+    DL_FOREACH(devh->streams, strmh) {
+        if (strmh->stream_if->bInterfaceNumber == interface_idx)
+            return strmh;
+    }
+
+    return NULL;
+}
+
+uvc_error_t
+usbhost_uvc_stream_open_ctrl(usbhost_uvc_device *devh, usbhost_uvc_stream_handle_t **strmhp,
+                             uvc_stream_ctrl_t *ctrl) {
+    /* Chosen frame and format descriptors */
+    usbhost_uvc_stream_handle_t *strmh = NULL;
+    usbhost_uvc_streaming_interface_t *stream_if;
+    uvc_error_t ret;
+
+    if (usbhost_uvc_get_stream_by_interface(devh, ctrl->bInterfaceNumber) != NULL) {
+        ret = UVC_ERROR_BUSY; /* Stream is already opened */
+        goto fail;
+    }
+
+    stream_if = usbhost_uvc_get_stream_if(devh, ctrl->bInterfaceNumber);
+    if (!stream_if) {
+        ret = UVC_ERROR_INVALID_PARAM;
+        goto fail;
+    }
+
+    strmh = (usbhost_uvc_stream_handle_t *) calloc(1, sizeof(*strmh));
+    if (!strmh) {
+        ret = UVC_ERROR_NO_MEM;
+        goto fail;
+    }
+    memset(strmh, 0, sizeof(usbhost_uvc_stream_handle_t));
+    strmh->devh = devh;
+    strmh->stream_if = stream_if;
+    //strmh->frame.library_owns_data = 1;
+
+
+    //TODO: Fix this
```

Comment:
What do you think this TODO could be referring to? I can't figure it out.

Also there's a fair bit of random commented out code in the function that we ought to just delete.

Reply:
Gone in 2bca5302b